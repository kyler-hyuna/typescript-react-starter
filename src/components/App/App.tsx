import React from 'react'
const Image = require('../../assets/image.jpg')
import value from '../../assets/names.json'

import * as actions from '@actions'
import { Store } from '@types'
import { connect, Dispatch } from 'react-redux'

interface ConnectedProps {
  counter: number
}

interface ConnectedDispatch {
  increment: () => void
}

type Props = ConnectedProps & ConnectedDispatch

export class App extends React.Component<Props> {
  render() {
    return (
      <>
        <h1>{this.props.counter} new react projects</h1>

        {value.names.map((a: string) => <p key={a}>{a}</p>)}

        <img src={Image} alt="" />

        <button onClick={this.props.increment}>Hey</button>
      </>
    )
  }
}

const mapStateToProps = (state: Store) => ({
  counter: state.counter,
})

const mapDispatchToProps = (dispatch: Dispatch<Store>) => ({
  increment: () => dispatch(actions.increment()),
})

export default connect<ConnectedProps, ConnectedDispatch, {}, Props>(
  mapStateToProps,
  mapDispatchToProps
)(App)
