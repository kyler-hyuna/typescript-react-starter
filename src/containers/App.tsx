import React from 'react'
const Image = require('../assets/image.jpg')
import value from '../assets/names.json'

interface Props {
  dog?: string
}

interface State {
  counter: number
}

class App extends React.Component<Props, State> {
  state = {
    counter: 0,
  }

  render() {
    return (
      <>
        <h1>{this.state.counter} new react projects</h1>

        {value.names.map((a: string) => <p key={a}>{a}</p>)}

        <img src={Image} alt="" />

        <button
          onClick={() => this.setState({ counter: this.state.counter + 1 })}
        >
          Hey
        </button>
      </>
    )
  }
}
export default App
