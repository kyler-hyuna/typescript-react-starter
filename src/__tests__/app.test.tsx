import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { create } from 'react-test-renderer'
import App from '../app'

describe('Main', () => {
  it('should work', () => {
    const tree = create(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
