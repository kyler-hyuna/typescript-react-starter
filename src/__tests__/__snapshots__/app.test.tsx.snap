// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should work 1`] = `
<div
  className="cover"
>
  <h1>
    Resource not found 
    <small>
      Error 404
    </small>
  </h1>
  <p
    className="lead"
  >
    The requested resource could not be found but may be available again in the future.
  </p>
</div>
`;
